src/main/java/uk/gov/ida/stub/idp/repositories/SessionRepositoryBase.java

```diff
@@ -0,0 +1,133 @@
+        Optional<String> sessionData = jdbi.withHandle(handle -> handle.select(
+                "select session_data from stub_idp_session where session_id = ?", sessionToken.toString())
+                .mapTo(String.class)
+                .findFirst());
+
+        if (!sessionData.isPresent()) {
+            return Optional.empty();
+        }
+
+        try {
+            String serializedSession = sessionData.get();
+            T session = objectMapper.readValue(unescapeJson(serializedSession.substring(1, serializedSession.length() - 1)), sessionType);
+
+            return Optional.of(session);
+        } catch (JsonParseException e) {
+            return cleanupSession(sessionToken);
+        } catch (JsonMappingException e) {
+            return cleanupSession(sessionToken);
+        } catch (IOException e) {
+            return cleanupSession(sessionToken);
+        }
+    }
+    
+    public SessionId updateSession(SessionId sessionToken, Session session) {
+        try {
+            String serializedSession = objectMapper.writeValueAsString(session);
+
+            jdbi.withHandle(handle ->
+                    handle.createUpdate(
+                            "UPDATE stub_idp_session SET session_data = to_json(:sessionData), last_modified = :lastModified WHERE session_id = :sessionId")
+                            .bind("sessionId", sessionToken.toString())
+                            .bind("sessionData", serializedSession)
+                            .bind("lastModified", Instant.now())
+                            .execute());
+        } catch (JsonProcessingException e) {
+            throw new SessionSerializationException("Unable to create session update.");
+        }
+
+        return sessionToken;
+    }
+    
+    
+    public void deleteSession(SessionId sessionToken) {
+        jdbi.withHandle(handle -> handle.execute("DELETE FROM stub_idp_session WHERE session_id = ?", sessionToken.toString()));
+    }
+
+    public Optional<T> deleteAndGet(SessionId sessionToken) {
+        Optional<T> session = get(sessionToken);
+        deleteSession(sessionToken);
+
+        return session;
+    }
+    
+    protected SessionId insertSession(SessionId sessionToken, T session) {
+        try {
+            String serializedSession = objectMapper.writeValueAsString(session);
+
+            jdbi.withHandle(handle ->
+                handle.createUpdate(
+                        "INSERT INTO stub_idp_session(session_id, session_data) VALUES (:sessionId, to_json(:sessionData))")
+                        .bind("sessionId", sessionToken.toString())
+                        .bind("sessionData", serializedSession)
+                        .execute());
+        } catch (JsonProcessingException e) {
+            throw new SessionSerializationException("Unable to create session update.");
+        }
+
+        return sessionToken;
+    }
+
+    private Optional cleanupSession(SessionId sessionToken) {
```

Comment:
If we are always going to return the same thing, we might as well make this method void